### ISSUE TITLE
Settings reset every few minutes

### ISSUE THREAD

**Issue by camaam** [form-o-fill/form-o-fill-chrome-extension#102]
I'm using Form-o-Filler for Chrome on macOS.

Remote rules import, Always show popup, etc. work fine after being set, but after about 3 minutes of inactivity the settings will be cleared out. The remote rules import URL becomes "undefined," checkboxes that were checked are cleared, etc. 

If you leave the options page open or the "Inspect views -> background page" inspector window open, the problem doesn't occur. It's only when the extension has been truly inactive for a few minutes.

I've tested this on three different Macbooks and they all appear to have the same issue.

Thanks

**Comment by thenoseman** [comment]
Hi,

I've been hunting this bug for a "long" time now. Thanks for the thorough debugging!

I'll try to debug it this week.

In the meantime can you please check if the rules file you are remote-importing contains a `"settings"` element? Like so:
```js
{
  "workflows": [
  ...
  ],
  "rules": { ... },
  "settings": { ...} # <--- THIS one
}
```

If so please *remove* that element (`"settings": { ...}`) from your remote import file and try again.

Please report back if the settings still reset after that change.

Thank you!

**Comment by camaam** [comment]
Thanks for the response! 

I tried that with a fresh install but it's still resetting.

I should note that it's resetting all the settings even if I don't have remote import enabled. For example, if I do a fresh install and only tick "Always Show Popup," that setting will be unticked after a few minutes (and the remote import input will read "undefined" even if I never touched it).

Thanks again!

**Comment by thenoseman** [comment]
Ok, so my first idea isn't what is causing this.
Can you try the dev-version of the extension please?
Here is a little how-to: https://github.com/form-o-fill/form-o-fill-chrome-extension/wiki#how-to-use-the-development-version-of-form-o-fill

Thanks!

**Comment by camaam** [comment]
The dev version appears to work! Thank you.

**Comment by thenoseman** [comment]
Well I'm not going to let you of that hook so fast :)
Is it ok if I come back to you if I find something worth double checking?

**Comment by camaam** [comment]
Of course, just let me know.

**Comment by thenoseman** [comment]
This is what is happening I think:
After ~15 seconds of inactivity the BG page is unloaded (.onSuspend event called).
Because I was too lazy I expect the settings in the state to always be loaded.
This is not the case so the options page resets the values.

Note to self:
- Use chrome.storage.local everywhere instead of localStorage.
- Wrap chrome.storage.* in promise via Storage class.
- See https://developer.chrome.com/apps/storage ("Overview")
- Might be able to use storage.sync.* some time to even sync rules/settings

**Comment by thenoseman** [comment]
Ok. I released 3.8.3 which hopefully fixes the unloading/settings issue. 

Can you please test it? Thanks!
